vault/resource_ad_secret_backend.go

```diff
@@ -0,0 +1,645 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func adSecretBackendResource() *schema.Resource {
+	fields := map[string]*schema.Schema{
+		"backend": {
+			Type:        schema.TypeString,
+			Default:     "ad",
+			ForceNew:    true,
+			Optional:    true,
+			Description: `The mount path for a backend, for example, the path given in "$ vault auth enable -path=my-ad ad".`,
+			StateFunc: func(v interface{}) string {
+				return strings.Trim(v.(string), "/")
+			},
+		},
+		"anonymous_group_search": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: `Use anonymous binds when performing LDAP group searches (if true the initial credentials will still be used for the initial connection test).`,
```

Comment:
Is the assumption that when default is not specified in the description for booleans, it defaults to false?

Reply:
That's correct. Also remember that the backend itself has defaults, so it's not always proper for Terraform to duplicate it, I think.